app/controllers/tasks_controller.rb

```diff
@@ -18,9 +18,33 @@ def create
         end
     end
 
+    def edit
+        @task = Task.find(params[:id])
+    end
+
+    def update
+        @task = Task.find(params[:id])
+    
+        if @task.update(task_params)
+          flash[:success] = "Task updated"
+          redirect_to root_url
+        else
+          render "edit"
+        end
+    end
+
+    def task_complete
```

Comment:
This doesn't look like a good name for an action, it will make your action helpers look weird...